apps/citizen-func/utils/redis_storage.ts

```diff
@@ -2,204 +2,70 @@ import * as E from "fp-ts/lib/Either";
-
-const isTrue = (b: boolean): b is true => b === true;
-
-const falsyResponseToError = (
-  response: E.Either<Error, boolean>,
-  error: Error
-): E.Either<Error, true> =>
-  pipe(
-    response,
-    E.filterOrElse(isTrue, () => error)
-  );
+import * as redis from "redis";
 
 export const setWithExpirationTask = (
-  redisClient: RedisClient,
+  redisClient: redis.RedisClientType,
   key: string,
   value: string,
   expirationInSeconds: number,
   errorMsg?: string
 ): TE.TaskEither<Error, true> =>
   pipe(
     TE.tryCatch(
+      () => redisClient.set(key, value, { EX: expirationInSeconds }),
       () =>
-        new Promise<E.Either<Error, true>>(resolve =>
-          redisClient.set(
-            key,
-            value,
-            "EX",
-            expirationInSeconds,
-            (err, response) =>
-              resolve(
-                falsyResponseToError(
-                  singleStringReply(err, response),
-                  new Error(
-                    errorMsg
-                      ? errorMsg
-                      : "Error setting key value pair on redis"
-                  )
-                )
-              )
-          )
-        ),
-      E.toError
+        new Error(errorMsg ? errorMsg : "Error setting key value pair on redis")
     ),
-    TE.chain(TE.fromEither)
+    TE.map(_ => true)
   );
 
 export const setTask = (
-  redisClient: RedisClient,
+  redisClient: redis.RedisClientType,
   key: string,
   value: string,
   errorMsg?: string
 ): TE.TaskEither<Error, true> =>
   pipe(
     TE.tryCatch(
+      () => redisClient.set(key, value),
       () =>
-        new Promise<E.Either<Error, true>>(resolve =>
-          // eslint-disable-next-line sonarjs/no-identical-functions
-          redisClient.set(key, value, (err, response) =>
-            resolve(
-              falsyResponseToError(
-                singleStringReply(err, response),
-                new Error(
-                  errorMsg ? errorMsg : "Error setting key value pair on redis"
-                )
-              )
-            )
-          )
-        ),
-      E.toError
+        new Error(errorMsg ? errorMsg : "Error setting key value pair on redis")
     ),
-    TE.chain(TE.fromEither)
+    TE.map(_ => true)
   );
 
 export const deleteTask = (
-  redisClient: RedisClient,
+  redisClient: redis.RedisClientType,
   key: string
-): TE.TaskEither<Error, boolean> =>
+): TE.TaskEither<Error, true> =>
   pipe(
     TE.tryCatch(
-      () =>
-        new Promise<E.Either<Error, boolean>>(resolve =>
-          redisClient.del(key, (err, response) =>
-            resolve(
-              falsyResponseToError(
-                integerReply(err, response),
-                new Error("Error deleting key value pair on redis")
-              )
-            )
-          )
-        ),
-      E.toError
+      () => redisClient.del(key),
+      () => new Error("Error deleting key value pair on redis")
     ),
-    TE.chain(TE.fromEither)
+    TE.map(_ => true)
```

Comment:
here you can use `constTrue` from `fp-ts/lib/function`

```suggestion
    TE.map(constTrue)
```

Reply:
Thanks, did it here [8b808bc](https://github.com/pagopa/io-messages/pull/12/commits/8b808bce649acdcfbf9202fcf0afdc6f93e6021c)